packages/widget_toolkit/lib/src/lib_search_picker/views/search_picker_page_with_dependencies.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import '../../../widget_toolkit.dart';
+import '../../base/models/item_builder.dart';
+import '../di/search_picker_dependencies.dart';
+
+class SearchPickerPageWithDependencies<T extends PickerItemModel>
+    extends StatelessWidget {
+  const SearchPickerPageWithDependencies({
+    super.key,
+    required this.title,
+    required this.hintText,
+    required this.retryText,
+    required this.service,
+    required this.loadingItemHeight,
+    this.itemBuilder,
+    this.selectedItem,
+    this.onItemTap,
+    this.errorBuilder,
+    this.emptyBuilder,
+    this.separatorBuilder,
+    this.showEmptyWidgetWhenNoResultsAreFound = true,
+  });
+
+  final String title;
+  final String hintText;
+  final String retryText;
+  final SearchPickerService<T> service;
+  final double loadingItemHeight;
+  final ItemPickerItemBuilder<T>? itemBuilder;
+  final T? selectedItem;
+  final Function(T?)? onItemTap;
+  final Widget Function(Exception)? errorBuilder;
+  final Widget Function()? emptyBuilder;
+  final Widget Function(int)? separatorBuilder;
+  final bool showEmptyWidgetWhenNoResultsAreFound;
+
+  @override
+  Widget build(BuildContext context) => MultiProvider(
```

Comment:
@nikolay-vasilev-prime here is one example whereisInTestMode should be integrated to enable the golden tests capabilities